backend/pkg/api/data_access/vdb_helpers_test.go

```diff
@@ -0,0 +1,388 @@
+package dataaccess
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/gobitfly/beaconchain/pkg/api/enums"
+	"github.com/gobitfly/beaconchain/pkg/commons/types"
+	"github.com/gobitfly/beaconchain/pkg/commons/utils"
+	"github.com/jmoiron/sqlx"
+	"github.com/lib/pq"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func setupConfig() {
+	cfg := types.Config{
+		Chain: types.Chain{
+			ClConfig: types.ClChainConfig{
+				EpochsPerSyncCommitteePeriod: 32,
+				AltairForkEpoch:              0,
+			},
+		},
+	}
+	utils.Config = &cfg
+}
+
+func setupTestDataAccess(t *testing.T) (da *DataAccessService, closeFunc func(), mock sqlmock.Sqlmock) {
+	setupConfig()
+
+	mockDB, mock, err := sqlmock.New()
+	assert.NoError(t, err)
+	sqlxDB := sqlx.NewDb(mockDB, "sqlmock")
+	dataAccess := &DataAccessService{
+		clickhouseReader: sqlxDB,
+		readerDb:         sqlxDB,
+		alloyReader:      sqlxDB,
+	}
+
+	return dataAccess, dataAccess.Close, mock
```

Comment:
TODO (non-blocking): should write unit tests for `dataAccess.Close`, can be done in another PR tho